src/contacts/ContactImporter.ts

```diff
@@ -69,6 +71,16 @@ export class ContactImporter {
 		)
 	}
 
+	// will check for permission and ask for it if it is not granted
+	async importContactsFromDeviceSafely() {
```

Comment:
we probably don't need to keep `importContactsFromDevice()` public.

Then this function can be named `importContactsFromDevice()` and the other one can be renamed, will make the interface a bit less clunky.